src/main/kotlin/no/nav/arbeidsplassen/importapi/exception/ImportApiErrorHandler.kt

```diff
@@ -67,16 +70,22 @@ private fun handleJsonProcessingException(error: JsonProcessingException): HttpR
         is JsonParseException -> HttpResponse
                 .badRequest(ErrorMessage("Parse error: at ${error.location}", PARSE_ERROR))
         is InvalidFormatException -> HttpResponse
-                .badRequest(ErrorMessage("Invalid value: ${error.value} at ${error.pathReference}", INVALID_VALUE))
-        is ValueInstantiationException -> HttpResponse.badRequest(ErrorMessage("Wrong value: ${error.message} at ${error.pathReference}", INVALID_VALUE))
+                .badRequest(ErrorMessage("Invalid value: ${error.value} at field ${feltFraPathReference(error.pathReference)}", INVALID_VALUE))
+        is ValueInstantiationException -> HttpResponse.badRequest(ErrorMessage("Wrong value: ${error.message} at field ${feltFraPathReference(error.pathReference)}", INVALID_VALUE))
         is InvalidNullException -> HttpResponse
                 .badRequest(ErrorMessage("Missing parameter: ${error.propertyName.simpleName}", MISSING_PARAMETER))
         is MismatchedInputException -> HttpResponse
-                .badRequest(ErrorMessage("Missing parameter: ${error.pathReference}", MISSING_PARAMETER))
+                .badRequest(ErrorMessage("Missing parameter: ${feltFraPathReference(error.pathReference)}", MISSING_PARAMETER))
         else -> HttpResponse.badRequest(ErrorMessage("Bad Json: ${error.localizedMessage}", UNKNOWN))
     }
 }
 
+fun feltFraPathReference(pathReference: String): String {
+    val regex = """\["(.*?)"]""".toRegex() // Matcher grupper i hakeparantes
+    val matches = regex.findAll(pathReference).map { it.groupValues[1] }.toList()
```

Comment:
Bare lurer på om den Arraylist->bolken i feilmeldingen brukt som eksempel kan være fra de som bruker batchinnending av ads..? Får man samme melding hvis man sender inn annonse via enkel innsending av annonse?

Reply:
Ja, ihvertfall i de tilfellene jeg har sett og prøvd:

```
fun extractLastBracketContent(input: String): String? {
    val regex = """\["(.*?)"]""".toRegex()
    val matches = regex.findAll(input).map { it.groupValues[1] }.toList()
    return matches.lastOrNull()
}

val inputs = listOf(
    "no.nav.arbeidsplassen.importapi.dto.AdDTO[\"reference\"]",
    "no.nav.arbeidsplassen.importapi.dto.AdDTO[\"title\"]",
    "Java.util.ArrayList[0]->no.nav.arbeidsplassen.importapi.dto.ContactDTO[\"title\"][\"Ball\"]"
)

inputs.forEach { input ->
    val result = extractLastBracketContent(input)
    println("Input: $input\nExtracted: $result\n")
}
```

```
Input: no.nav.arbeidsplassen.importapi.dto.AdDTO["reference"]
Extracted: reference

Input: no.nav.arbeidsplassen.importapi.dto.AdDTO["title"]
Extracted: title

Input: Java.util.ArrayList[0]->no.nav.arbeidsplassen.importapi.dto.ContactDTO["title"]["Ball"]
Extracted: Ball
```

Reply:
Det er en forbedring likevel! Tanken med TRello-oppgaven var å gjøre det lettere for brukere av import-apiet å implementere feilmeldinger til brukerne, da vi har opplevd at arbeidsgivere ikke forstår hvorfor det feiler når de bruker SAP/Visma osv osv. Forståelig nok med den feilmeldingen vi gir. Her må det matches på streng, men det er hvert fall en forståelig streng som ikke gir masse pakkeinformasjon..